### PR TITLE
move from flannel to openshiftsdn

### PR DESCRIPTION
**Pull request by joelddiaz**
openshiftsdn is now the default: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/installconfig.go#L82

revendor installer

this fixes the masters coming up not ready

### CODE REVIEW

**Comment by csrwng** [comment]
@joelddiaz @dgoodwin is there a way to get the default from the installer code so we don't have to maintain this ourselves?

File: pkg/install/convertconfig.go [hunk 1]
```diff
@@ -100,8 +100,7 @@ func generateInstallConfig(cd *hivev1.ClusterDeployment, adminPassword, sshKey,
 		},
 		BaseDomain: spec.Config.BaseDomain,
 		Networking: types.Networking{
-			// TODO: installer currently only works with flannel, flagged for TODO on their end here: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/installconfig.go#L82
-			Type: "flannel",
+			Type: "OpenshiftSDN",
```
Comment (dgoodwin) [inline pkg/install/convertconfig.go]: I hardcoded as this was all they supported at the time, now I think we can remove the hardcode, and instead use what convertNetworkPlugin returns, see the commented out code at the top of this function. Bring that back to life and then we can just pass through whatever the user specified.

**Comment by dgoodwin** [comment]
Nothing yet, shared validation/defaulting was something I had wondered about though.

**Comment by joelddiaz** [comment]
k, revendored the installer to pull in the new sdn types and made the updates as needed

File: pkg/install/convertconfig.go [hunk 2]
```diff
@@ -34,28 +37,30 @@ import (
...
+	networkType, err := convertNetworkingType(spec.Config.Networking.Type)
+	if err != nil {
+		return nil, err
+	}
+
 	platform := types.Platform{}
 	if spec.Config.Platform.AWS != nil {
 		aws := spec.Config.Platform.AWS
-		platform.AWS = &types.AWSPlatform{
+		////platform.AWS = &types.AWSPlatform{
```
Comment (dgoodwin) [inline pkg/install/convertconfig.go]: Just some extraneous comments here and below.

**Comment by dgoodwin** [comment]
Fixes: https://github.com/openshift/hive/issues/82

**Comment by joelddiaz** [comment]
for real this time...test cases updated

**Pull request merged**